### PR TITLE
MaxAmmo Setter && NightVision should also be true when NightVision

### PR DESCRIPTION
**Pull request by louis1706**
- Fix Custom Weapons can reload without going through the animation
- Added MaxAmmo Setter
- Added CanSeeThroughDark
- Added NightVisionEnabled

### CODE REVIEW

File: Exiled.API/Features/Items/Firearm.cs [hunk 1]
```diff
@@ -118,9 +118,27 @@ public byte Ammo
...
+            {
+                switch (Base.AmmoManagerModule)
+                {
+                    case TubularMagazineAmmoManager tubularMagazineAmmoManager:
+                        tubularMagazineAmmoManager.MaxAmmo = value;
+                        break;
+                    case ClipLoadedInternalMagAmmoManager clipLoadedInternalMagAmmoManager:
+                        clipLoadedInternalMagAmmoManager.MaxAmmo = value;
+                        break;
+                    case AutomaticAmmoManager automaticAmmoManager:
```
Comment (Thundermaker300) [inline Exiled.API/Features/Items/Firearm.cs:136]: This switch should have a default clause and throw an exception if it's a module that cannot be set.

File: Exiled.API/Features/Items/Firearm.cs [hunk 2]
```diff
@@ -138,9 +156,10 @@ public byte Ammo
         public bool Aiming => Base.AdsModule.ServerAds;
 
         /// <summary>
-        /// Gets a value indicating whether the firearm's flashlight module is enabled.
+        /// Gets a value indicating whether the firearm's flashlight module is enabled or if playing see trough a NightVision.
         /// </summary>
-        public bool FlashlightEnabled => Base.Status.Flags.HasFlagFast(FirearmStatusFlags.FlashlightEnabled);
+        public bool FlashlightEnabled => Base.Status.Flags.HasFlagFast(FirearmStatusFlags.FlashlightEnabled) ||
```
Comment (Thundermaker300) [inline Exiled.API/Features/Items/Firearm.cs:161]: Separate property for night vision

File: Exiled.API/Features/Items/Firearm.cs (hunk 1 above)
Comment (louis1706) [reply Exiled.API/Features/Items/Firearm.cs:136]: The only default would be with Disruptor than MaxAmmo is actually useless

Comment (Thundermaker300) [reply Exiled.API/Features/Items/Firearm.cs:136]: I'm aware, which is why it should throw an error as opposed to not doing anything, as you cannot set the max ammo in this case.

File: Exiled.API/Features/Items/Firearm.cs (hunk 2 above)
Comment (louis1706) [reply Exiled.API/Features/Items/Firearm.cs:161]: How would you call the property that combined both ?

Comment (Thundermaker300) [reply Exiled.API/Features/Items/Firearm.cs:161]: Separate property for Flashlight vs night vision

Comment (louis1706) [reply Exiled.API/Features/Items/Firearm.cs:161]: > Separate property for Flashlight vs night vision

I still want a property than combine both of them how should i call it ?

Comment (Thundermaker300) [reply Exiled.API/Features/Items/Firearm.cs:161]: IsEmittingLight would probably work

File: Exiled.API/Features/Items/Firearm.cs [hunk 3]
```diff
@@ -143,6 +165,16 @@ public byte Ammo
         /// </summary>
         public bool FlashlightEnabled => Base.Status.Flags.HasFlagFast(FirearmStatusFlags.FlashlightEnabled);
 
+        /// <summary>
+        /// Gets a value indicating whether the firearm's flashlight module is enabled or if playing see trough a NightVision.
```
Comment (Misaka-ZeroTwo) [inline Exiled.API/Features/Items/Firearm.cs:169]:
```suggestion
        /// Gets a value indicating whether the firearm's NightVision is being used.
```

File: Exiled.API/Features/Items/Firearm.cs [hunk 4]
```diff
@@ -143,6 +165,16 @@ public byte Ammo
...
+        /// </summary>
+        public bool NightVisionEnabled => Aiming && Base.HasAdvantageFlag(AttachmentDescriptiveAdvantages.NightVision);
+
+        /// <summary>
+        /// Gets a value indicating whether the firearm's flashlight module is enabled or if playing see trough a NightVision.
```
Comment (Misaka-ZeroTwo) [inline Exiled.API/Features/Items/Firearm.cs:174]:
```suggestion
        /// Gets a value indicating whether the firearm's flashlight module is enabled or NightVision is being used.
```

File: Exiled.API/Features/Items/Firearm.cs [hunk 5]
```diff
@@ -143,6 +165,16 @@ public byte Ammo
...
+        /// </summary>
+        public bool IsEmittingLight => FlashlightEnabled || NightVisionEnabled;
```
Comment (louis1706) [inline Exiled.API/Features/Items/Firearm.cs:176]:
```suggestion
        public bool CanSeeThroughDark => FlashlightEnabled || NightVisionEnabled;
```

**Comment by Misaka-ZeroTwo** [comment]
Tested by trouble.

**Pull request merged**